src/app_model/backends/qt/_qmenu.py

```diff
@@ -305,7 +305,10 @@ def _rebuild(
     exclude: Collection[str] | None = None,
 ) -> None:
     """Rebuild menu by looking up `menu` in `Application`'s menu_registry."""
-    menu.clear()
+    actions = menu.actions()
+    for action in actions:
+        menu.removeAction(action)
```

Comment:
interesting. are both of these required to be able to use `create(parent=...)` here?   (i.e. *not* using clear as well as using `setParent()` inside of create?)

happy to go with whatever works best.  just curious

Reply:
We need to keep reference to actions to prevent them being garbage collected, but we need to remove them from the menu before rebuild.

Reply:
aha... good one